core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -605,6 +606,75 @@ public FileSystemMasterView getFileSystemMasterView() {
   }
 
   /**
+   * Checks the consistency of the files and directories in the subtree under the path.
+   *
+   * @param path the root of the subtree to check
+   * @param options the options to use for the checkConsistency method
+   * @return a list of paths in Alluxio which are not consistent with the under storage
+   * @throws FileDoesNotExistException if the path does not exist
+   * @throws InvalidPathException if the path is invalid
+   * @throws IOException if an error occurs interacting with the under storage
+   */
+  public List<AlluxioURI> checkConsistency(AlluxioURI path, CheckConsistencyOptions options)
+      throws FileDoesNotExistException, InvalidPathException, IOException {
+    List<AlluxioURI> inconsistentUris = new ArrayList<>();
+    try (LockedInodePath parent = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ);
+        InodeLockList children = mInodeTree.lockDescendants(parent, InodeTree.LockMode.READ)) {
+      if (!checkConsistencyInternal(parent.getInode(), parent.getUri())) {
+        inconsistentUris.add(parent.getUri());
+      }
+      for (Inode child : children.getInodes()) {
+        AlluxioURI currentPath = mInodeTree.getPath(child);
+        if (!checkConsistencyInternal(child, currentPath)) {
+          inconsistentUris.add(currentPath);
+        }
+      }
+    }
+    return inconsistentUris;
+  }
+
+  /**
+   * Check if a path is consistent between Alluxio and the underlying storage.
+   *
+   * A path without a backing under storage is always consistent.
```

Comment:
Its not possible at the moment, but there is some logic in `MountTable` which returns a null ufs if it cannot find a mount point for the path. I can remove this if you think its confusing.